add simple meson.build file

brainpower opened the pull request:
it includes all targets the Makefiles had at this point.

Since I had some time at my hands, was a little bored and thought, "this should be very easy to port to meson" when I read the Makefiles, I wrote a meson build file as mentioned in issue #18 

One question arose though: 
Does ashuffle require a specific (minimum) version of libmpdclient ?

The Makefiles did not care, but it would be trivial to add a version constraint to this...
See first example at: https://mesonbuild.com/Dependencies.html

joshkunz commented on meson.build:
```diff
@@ -0,0 +1,34 @@
+)
+
+
+run_target( 'gdb',
```
This causes some weird build errors for me if `gdb` is not present. It would probably be best to just remove it. I only added it to have a quick shortcut for debugging :)

Errors:
```
/ashuffle # meson build
The Meson build system
Version: 0.48.2
Source dir: /ashuffle
Build dir: /ashuffle/build
Build type: native build
Project name: ashuffle
Project version: 1.0.3
Native C compiler: cc (gcc 8.2.0 "cc (Alpine 8.2.0) 8.2.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.5.4)
Dependency libmpdclient found: YES 2.16
Program gdb found: NO

meson.build:32:0: ERROR:  Tried to use non-existing executable 'nonexistingprogram'

A full log can be found at /ashuffle/build/meson-logs/meson-log.txt
```

brainpower replied:
oh, damn, missed that... I had thought it would only complain if you'd try to use the target...
well, one still can do `ninja && gdb ./ashuffle` ^^

joshkunz commented on meson.build:
```diff
@@ -0,0 +1,34 @@
+
```
```suggestion
```

brainpower replied:
done

joshkunz commented on meson.build:
```diff
@@ -0,0 +1,34 @@
+
+gdb = find_program('gdb', required: false)
+
+
```
It would be nice to have more uniform spacing here. Let’s say one space?

brainpower replied:
gdb stuff was removed anyway

joshkunz commented:
Hey neat thanks for pull request! This is really nice to have, since it seems to be the way that the MPD ecosystem is moving (and it’s much nicer than autoconf!). Would you mind also updating the build instructions in the readme? I’ll remove the Makefile during the next backwards incompatible change.

I’ve also sent out a small review, with comments on the changes.

Thanks again for the contribution!

brainpower commented:
Well... almost any "newer" build system out there is nicer than autoconf. 
That one is just a very big mess and very hard to learn... 
I tried at some point and gave up after reading the first chapter of the tutorial because I thought it way too complicated for the problem I was trying to solve (building a small project like ashuffle is)
I switched to CMake then (meson didn't exist yet). In my opinion it's syntax is unnecessarily verbose, though. I'm glad meson managed to avoid that.

I really like how simple the meson.build file for some small project like this can be.

Back to topic:
* switched to spaces (forgot to :set expandtab :( )
* removed empty lines
* removed the gdb stuff
